interface/cpplog.hpp

```diff
@@ -0,0 +1,1256 @@
+#pragma once
```

Comment:
Where did this file come from? It seems like unneeded bloat

Reply:
I can take this file out, but I wanted to put some logging similar to the logging library in python. The choice of this library is completely arbitrary, but I think it would be good to have some sort of ability to have logging. The only place i have it implemented right now is in choosing the selection. Most analysis codes don't use this variable at all and the code crashes if you don't have it. I think having a Warning, not a full blow error would be good for checking.

Reply:
If you have idea of how to do this, I'd love to use that! or if you think the extra library is bloat in general, we will need to offload all of these checks on the python code or something i think to make it easier to make checks

Reply:
Isn't there something in Boost? It's already distributed in CMSSW. In general I'd prefer to use a library rather than copying some other code.

Reply:
The boost library is trivial I think? I tried it but it had some errors compiling, but I'll give it another try

Reply:
It least compiles with the includes added. I have never used it but I think that's a much better option than adopting a random dependency. 

As for selection I am also looking at making this optional.

Reply:
Boost library still gives a lot of troubling in the linking. I ended up going with a library in CMSSW that it seems to work fine enough though it doesn't have the fancy line number technology of the previous library.

Selection should be optional now. For the fake rate (doesn't use selection since histograms are defined in the file), it works fine.